### PR TITLE
Added support for gitlab

### PR DESCRIPTION
**Pull request by nirzari**

### CODE REVIEW

File: reviewrot/gitlabstack.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                    log.debug("No projects found for user/group name %s",
+                              user_name)
+                for project in projects:
+                    res = self.get_reviews(uname=user_name, project=project,
+                                           state_=state_, value=value,
+                                           duration=duration)
+                    response.append(res)
+        return response
+
+    def get_reviews(self, uname, project, state_=None,
```
Comment (amisstea) [inline reviewrot/gitlabstack.py]: Almost everything in this method is identical to what you have in the `GithubService` class. It is also repeated in `PagureService` too. Can you abstract it into the base class?

Comment (nirzari) [inline reviewrot/gitlabstack.py]: @amisstea I will come back to address 'replication' issue later on as we discussed. Could you please review the remaining functionality. Thanks.

File: reviewrot/gitlabstack.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            try:
+                # get project object for given repo_name(project name)
+                project = gl.projects.get(os.path.join(user_name, repo_name))
+            except GitlabGetError:
+                log.exception('Project %s not found for user %s',
+                              repo_name, user_name)
+                raise Exception('Project %s not found for user %s'
+                                % (repo_name, user_name))
+            # get merge requests for specified username and project name
+            res = self.get_reviews(uname=user_name, project=project,
```
Comment (amisstea) [inline reviewrot/gitlabstack.py]: You're not doing anything with `res` after this. Should it be appended to `response`?

File: reviewrot/gitlabstack.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            res_ (list): Returns list of pull requests for specified
+                         user(group) name and project name
+        """
+        log.debug('Looking for merge requests for %s -> %s',
+                  uname, project.name)
+
+        # get list of open merge requests for a given repository(project)
+        merge_requests = project.mergerequests.list(project_id=project.id,
+                                                    state='opened')
+        if not merge_requests == 0:
```
Comment (amisstea) [inline reviewrot/gitlabstack.py]: Is this correct? Shouldn't it be:

```python
if merge_requests == 0:
```

File: reviewrot/gitlabstack.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+                comments.append('%s %s %s' % (', with',
+                                              mr.user_notes_count,
+                                              'comments'))
+            comments = ' '.join(comments)
+            # format and print the resultant pull request string
+            res = GitlabReview(user=str(mr.author.username),
+                               title=str(mr.title),
+                               url=str(mr.web_url),
+                               time=time,
+                               comments=comments)
```
Comment (amisstea) [inline reviewrot/gitlabstack.py]: I think the use of `comments` is a bit complex for what it contains (it also doesn't need to be a list when you first create it). All you are effectively doing is formatting the number of comments into a user friendly string. I believe that should live inside `BaseReview.__str__`. You could just pass `mr.user_notes_count` to the `comments` arg of `GitlabReview`. This change should be reflected across all other classes extending from `BaseService`.

File: reviewrot/gitlabstack.py (hunk 2 above)
Comment (nirzari) [inline reviewrot/gitlabstack.py]: yes, must have been deleted that line while moving some parts.

File: reviewrot/gitlabstack.py (hunk 3 above)
Comment (nirzari) [inline reviewrot/gitlabstack.py]: Yeah I was just changing it from "if len(merge_requests) == 0" to "if not merge_requests". Forgot to take out '== 0' part in rush. Thanks

File: reviewrot/gitlabstack.py (hunk 4 above)
Comment (nirzari) [inline reviewrot/gitlabstack.py]: Okay, I will try to move it to __str__

File: reviewrot/gitlabstack.py (hunk 1 above)
Comment (nirzari) [inline reviewrot/gitlabstack.py]: I was able to move most of stuff from that method to base class. I will update all respective files.

**Pull request closed without merging**